packages/expo-router/src/getLinkingConfig.ts

```diff
@@ -2,45 +2,54 @@ import { getActionFromState, LinkingOptions } from '@react-navigation/native';
 
 import { RouteNode } from './Route';
 import { State } from './fork/getPathFromState';
-import { getReactNavigationConfig, Screen } from './getReactNavigationConfig';
+import { getReactNavigationConfig } from './getReactNavigationConfig';
 import {
   addEventListener,
   getInitialURL,
   getPathFromState,
   getStateFromPath,
 } from './link/linking';
 
-export function getNavigationConfig(
-  routes: RouteNode,
-  metaOnly: boolean = true
-): {
-  initialRouteName?: string;
-  screens: Record<string, Screen>;
-} {
+export function getNavigationConfig(routes: RouteNode, metaOnly: boolean = true) {
   return getReactNavigationConfig(routes, metaOnly);
 }
 
-export type ExpoLinkingOptions = LinkingOptions<object> & {
+export type ExpoLinkingOptions<T extends object = Record<string, unknown>> = LinkingOptions<T> & {
   getPathFromState?: typeof getPathFromState;
 };
 
-export function getLinkingConfig(routes: RouteNode, metaOnly: boolean = true): ExpoLinkingOptions {
+export function getLinkingConfig(
+  routes: RouteNode,
+  overrides: Partial<ExpoLinkingOptions> = {},
+  metaOnly: boolean = true
+): ExpoLinkingOptions {
+  // Returning `undefined` / `null from `getInitialURL` are valid values, so we need to track if it's been called.
+  let hasCachedInitialUrl = false;
+  let initialUrl: ReturnType<typeof getInitialURL> | undefined;
+
   return {
-    prefixes: [],
-    // @ts-expect-error
+    prefixes: overrides.prefixes ?? [],
```

Comment:
Hi @marklawlor, first off thanks for implementing this! It'll be quite helpful for #27163.

I wanted to check around the functionality for `prefixes`, since this particular config value should help me solve https://github.com/appcues/appcues-react-native-module/issues/147. It appears that currently prefix values are ignored in the fork of some react-navigation code. This was fine as prefixes was always empty, but with this change that won't always be the case.

Compare react navigation:

https://github.com/react-navigation/react-navigation/blob/a41d932198d1b48d84f410be1633fc5ddcb2df3f/packages/native/src/useLinking.native.tsx#L123-L136

vs the expo fork:

https://github.com/expo/expo/blob/4b34b0c955903403529d13d2dc8ddae5c5f7e098/packages/expo-router/src/fork/useLinking.native.ts#L125-L134

The react navigation implementation passes the prefixes ref to `extractPathFromURL` whereas `extractExpoPathFromURL` doesn't account for the prefixes.